### DISCUSSION TITLE
Performance decreases rapidly when increasing #images

### DISCUSSION THREAD

**Discussion by neriniccolodei** [OpenStitching/stitching#177] [General]
Hello,

Thank you once again for your precious package and support. 

I have noticed that the algorithm's performance decreases rapidly when trying to stitch together multiple images. For example, at around 80 frames I have experienced that panoramas are generated incorrectly, merging the last few frames (e.g., 10 frames) at the beginning of the panorama, instead of the natural end. 

Moreover, the system just cannot process a number of images such as, e.g., 120 (even on a powerful PC).

I was curious to ask if you've encountered a similar issue and how to approach it, if there's anything (maybe optimisation) that can be tuned to try out if the problem is solved... 

Kind regards

**Comment by lukasalexanderweber** [comment]
It's already reported [here](https://stackoverflow.com/questions/22056737/opencv-stitching-images-from-a-grid-of-images) and in one or two discussions here, that the system gets to a point of instability. Do you use the range matcher? This will limit the images that are compared if you can pass the images in the correct order. Have you used verbose mode to see where/when the workflow is stuck. Maybe you could implement a logging with some timers. Have you tried breaking down your Image Set into subproblems (stitching subregions of your imgages and puzzle them together at a smaller scale)?
